File: labs/lab_3/day3php/models/forms/lab_two_form_validation.php
<?php


/*
Doc Title  Lab_two_form_validation.php
Author: Hugh Ybarra
Description: Validates all fields of the lab 2 form
form method: post
example:
first
last
date
number
currency
url
checkbox
radios

validates form and returns a json encoded doc with true or false
if the form is valid or invalid
 
 */
 
class Validate_lab_Two_Forms{
	
	public function validate_forms($form_data){
		
		// Declaring Variables
		$first_name 	= "";
		$last_name 		= "";
		$email 			= "";
		$date 			= "";
		$number 		= "";
		$currency 		= "";
		$url 			= "";
		$select 		= "";
		$checkbox		= "";
		$gender			= "";
		$terms_agree 	= "";
		$mail_list 		= "";
		$response 		= "";
		$is_valid 		= "";
		
		// Declaring Regex Variables
		$email_regex 		= "/[^@]+@[^@]+\.[^@]+/";
		$password_regex 	= "/^[a-zA-Z0-9]{8,15}/";
		
		$string_regex 		= "/^[a-zA-Z]{3,15}$/";
		
		$number_regex 		= "(-?\d+(?:\.\d+)?+)";
		// using filter validate instead
		//FILTER_VALIDATE_URL
		// $url_regex 			= "/ ^http\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,3}(\/\S*)?$ /";
		
		
		// check for the first name 
		//===================================================
		if (!empty($form_data["first_name"])){
			// Regex validation
			if (preg_match($string_regex, $form_data["first_name"])){
				$first_name = $form_data["first_name"];
			}else{
				$first_name = False;
			}
		}else {
			$first_name = False;
		}// end of first name 
		
		// check for the last name 
		//===================================================
		if (!empty($form_data["last_name"])){
			// Regex validation
			if(preg_match($string_regex, $form_data["last_name"])){
				$last_name = $form_data["last_name"];
			}else{
				$last_name = False;
			}
		}else{
			$last_name = False;
		}// end last name
		
		// check for the date
		//===================================================
		if (!empty($form_data["form_date"])){
			$date = $form_data["form_date"];
		}else{
			$date = False;
		}// end date
		
		// check for the Number
		//===================================================
		// echo gettype($form_data["form_number"]);
		if(!empty($form_data["form_number"])){
			if(preg_match($number_regex ,$form_data["form_number"])){
				$number = $form_data["form_number"];
			}else{
				$number = False;
			}
		}else{
			$number = False;
		}// end number
		
		// check for the currency
		//===================================================
		if(!empty($form_data["form_currency"])){
			if(preg_match($number_regex, $form_data["form_currency"])){
				$currency = $form_data["form_currency"];
			}else{
				$currency = False;
			}
		}else{
			$currency = False;
		}// end currency
		
		// check for the url
		//===================================================
		if(!empty($form_data["form_url"])){
			if(filter_var($form_data["form_url"], FILTER_VALIDATE_URL)){
				$url = $form_data["form_url"];
			}else{
				$url = False;
			}
		}else{
			$url = False;
		}// end url
		
		// check the select
		//===================================================
		if(!empty($form_data["form_select"])){
			$select = $form_data["form_select"];
		}else{
			$select = False;
		}// end for select
		
		// check for gender
		//===================================================
		if(!empty($form_data["sex"])){
			$gender = $form_data["sex"];
		}else{
			$gender = False;
		}// end gender 
		// check for check boxes
		//===================================================
		if(!empty($form_data["agree"])){
			$terms_agree = "agree";
		}else{
			$terms_agree = False;
		}
		
		if(!empty($form_data["form_mailing_list"])){
			$mail_list = $form_data["form_mailing_list"];
		}else{
			$mail_list = False;
		}
		// end checkbox 
		//===================================================
		
		// check all fields 
		//===================================================
		if ($first_name && $last_name && $date && $number && $currency && $url && $select && $gender){
			$is_valid = True;
		}else{
			$is_valid = False;
		}
		
		// json encode data
		//===================================================
		$form_data_array = array(
			"success" => $is_valid,
			"first_name" => $first_name,
			"last_name"=> $last_name,
			"date" => $date,
			"number" => $number,
			"currency" => $currency,
			"url" => $url,
			"select" => $select,
			"gender" => $gender,
			"terms_agree" => $terms_agree,
			"mailing_list"=> $mail_list	
		);
		echo json_encode($form_data_array);
		return json_encode($form_data_array);
		
	}// end validate_forms function
}// end validate_lab_two_form class

File: labs/lab_2/day2Python/models/forms/lab_two_form_validation.py
#!/usr/bin/python
__author__ = 'Hugh'

'''
Doc Title  Lab_two_form_validation.py
Author: Hugh Ybarra
Description: Validates all fields of the lab 2 form
form method: post
example:
first
last
date
number
currency
url
checkbox
radios

validates form and returns a json encoded doc with true or false
if the form is valid or invalid
'''


class Validate_Lab_Two_Form:
    def __init__(self):
        self.base_path = "/Applications/MAMP/htdocs/SSL/day2Python/models/forms"

    def validate_form(self, my_url):
        #document imports
        import re
        import json
        #=======================================
        #=======================================
        #DELETE THIS WHEN DONE TESTING
        import models.view
        from models.view import View
        view_model = View()
        view_model.print_header()
        # DELETE THIS WHEN DONE TESTING
        #=======================================
        #=======================================

        #declaring validator variables
        first_name  = ""
        last_name   = ""
        email       = ""
        date        = ""
        number      = ""
        currency    = ""
        url         = ""
        select      = ""
        checkbox    = ""
        sex         = ""
        user_data   = ""
        sex         = ""
        terms_agree = ""
        mail_list   = ""
        response    = {}
        is_valid    = True
        #declaring regex variables
        email_regex = "[^@]+@[^@]+\.[^@]+"
        #look into finding a better regex for this email
        #it works for the assignmetn but!
        #i am probably going to have to validate their email address by emailing a link to them then have them click on it
        # so a basic validator works for now.
        # find better email regex !

        password_regex  = "^[a-zA-Z0-9]{8,15}"
        string_regex    = "^[a-zA-Z]{3,}"
        number_regex    = "^[0-9]"
        # url_regex       = "^[a-zA-Z0-9\-\.]+\.(com|org|net|mil|edu|COM|ORG|NET|MIL|EDU)$"
        url_regex       = "^http\://[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,3}(/\S*)?$"

        # check for the first name
        #===================================================
        if my_url.getvalue("first_name"):
            #regex validation
            match_object = re.search(string_regex, my_url.getvalue("first_name"))
            # match_object = re.search(string_regex, my_url.getvalue("firstName"))
            if match_object:
                    first_name = my_url.getvalue("first_name")
            else:
                first_name = False
        else:
            first_name = False

        # check for the last name
        #===================================================
        if my_url.getvalue("last_name"):
            #regex validation
            match_object = re.search(string_regex, my_url.getvalue("last_name"))

            if match_object:
                last_name = my_url.getvalue("last_name")
            else:
                last_name = False
        else:
            last_name = False
        # check for the date
        #===================================================
        if my_url.getvalue("form_date"):
            date = my_url.getvalue("form_date")
        else:
            date = False
        # check for Number
        #===================================================
        if my_url.getvalue("form_number"):
            #regex validation
            match_object = re.match(number_regex, my_url.getvalue("form_number"))
            if match_object:
                number = my_url.getvalue("form_number")
            else:
                number = False
        else:
            number = False
        # check for currency
        #===================================================
        if my_url.getvalue("form_currency"):
            #regex validation
            match_object = re.search(number_regex, my_url.getvalue("form_currency"))
            if match_object:
                currency = my_url.getvalue("form_currency")
            else:
                currency = False
        else:
            currency = False
        # check for url
        #===================================================
        if my_url.getvalue("form_url"):
            #regex validation
            match_object = re.search(url_regex, my_url.getvalue("form_url"))
            if match_object:
                url = my_url.getvalue("form_url")
            else:
                url = False
        else:
            url = False
        # check for select
        #===================================================
        if my_url.getvalue("form_select"):
            select = my_url.getvalue("form_select")
        else:
            select = False
        #check for sex
        #===================================================
        if my_url.getvalue("sex"):
            sex = my_url.getvalue("sex")
        else:
            sex = False
        #check for agree
        #===================================================
        if my_url.getvalue("agree"):
            terms_agree = my_url.getvalue("agree")
        else:
            terms_agree = False
        #check for mailing list
        #===================================================
        if my_url.getvalue("form_mailing_list"):
            mail_list = my_url.getvalue("agree")
        else:
            mail_list = False

        #end of field validation
        #====================================================
        #====================================================
        #====================================================
        # print first_name, "</br><hr>", \
        # last_name, "</br><hr>", \
        # date, "</br><hr>",\
        # number, "</br><hr>",\
        # currency, "</br><hr>",\
        # url,"</br><hr>",\
        # select,"</br><hr>",\
        # sex, "</br><hr>"



        if first_name and last_name and url and date and number and currency and select and sex:
            form_valid = True
        else:
            form_valid = False

        form_object = {
            "success":form_valid,
            "first_name":first_name,
            "last_name":last_name,
            "date":date,
            "number":number,
            "currency": currency,
            "url":url,
            "select":select,
            "sex":sex,
            "terms_agree":terms_agree
        }


        if form_valid:
            print json.dumps(form_object)
        else:
            print json.dumps(form_object)